### ISSUE TITLE
BUG: Tests on GitHub fail because of bug in latest version of Xugrid

### ISSUE THREAD

**Issue by Tjalling-dejong** [Deltares/hydromt#332]
### HydroMT version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have checked that this bug exists on the latest version of HydroMT.   

- [X] I have checked that this bug exists on the main branch of HydroMT.            

### Reproducible Example

Error occurs when pushing into main

### Issue Description

@alimeshgi has updated the issue template yamls and pushed it into main. The tests fail because there is in error in conftest.py here:

```
@pytest.fixture
    def demuda():
        import xugrid as xu
    
>       uds = xu.data.adh_san_diego()

```

The hash of the download of the San Diego dataset does not match with the known hash, indicating a corrupted file or outdated hash:

```
ValueError: SHA256 hash of downloaded file (ADH_SanDiego.nc) does not match the known hash: expected 50f595d4a2ab84a42b22ccfa4e3a0c88a3fa6a059ad408444eeed76143a9e17b but got c09eb694a0951f5d718179ebb167d1e7c2fdb8234ba8a5bb588dae3a3610b76a. Deleted download for safety. The downloaded file may have been corrupted or the known hash may be outdated.
```

I could not reproduce the error locally with the latest version of Xugrid.

### Installed Versions

Latest version on GitHub

**Comment by hboisgon** [comment]
xugrid downloads the data in a cache xugrid folder without version control. So if you already downloaded the data locally, this step is skipped.

**Comment by hboisgon** [comment]
I see xugrid released 0.3.0 about 2 months ago and this release removes the dependency on pygeos so we fully remove that in hydromt too.
However Huite updated 2 weeks ago adh_san_diego data causing the hash to change. This was not released yet but even for a fixed version of xugrid, the download is done in the main dev branch rather than the tagged release branch causing our error. 

So more or less solutions:

- wait for a xugrid release
- stop using this dataset in our test

In any case, I'll open an issue in xugrid repository.
